viewer/packages/pointclouds/src/potree-three-loader/tree/PointCloudOctree.ts

```diff
@@ -145,21 +146,26 @@ export class PointCloudOctree extends PointCloudTree {
     bbRoot.children = visibleBoxes;
   }
 
-  updateMatrixWorld(force: boolean): void {
-    if (this.matrixAutoUpdate === true) {
-      this.updateMatrix();
-    }
+  updateMatricesForDescendants(): void {
+    this.traverseVisible(node => node.matrixWorld.multiplyMatrices(this.matrixWorld, node.matrix));
+  }
 
+  /**
+   * Override updateMatrixWorld to not update children recursively. Child transformations
+   * are defined in relation to the base octree, and are instead updated using updateMatricesfordescendants()
+   */
+  override updateMatrixWorld(force: boolean): void {
```

Comment:
Isn't `Object3D.updateWorldMatrix(false, true)` will do the same out of the box? [Link](https://threejs.org/docs/index.html?q=object#api/en/core/Object3D.updateWorldMatrix) to threejs docs.

Reply:
The implementation here is based on that in the [source for Object3D](https://github.com/mrdoob/three.js/blob/309b00afb6dcbc5e6c58e72f10eaa8d2e8888c83/src/core/Object3D.js#L601) , but with the important difference that it updates the children like a hierarchy, so each child gets the transformation of its parent. But that's apparently not how Potree does things, so I instead just manually apply the matrix to each child in the `updateMatricesForDescendants` function. That`s why it was manually implemented before as well, instead of just calling the `Object3D` implementation.